streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -680,6 +674,14 @@ private KafkaStreams(final InternalTopologyBuilder internalTopologyBuilder,
         }
         final ProcessorTopology globalTaskTopology = internalTopologyBuilder.buildGlobalStateTopology();
         final long cacheSizePerThread = totalCacheSize / (threads.length + (globalTaskTopology == null ? 0 : 1));
+        final boolean createStateDirectory = taskTopology.hasPersistentLocalStore() ||
```

Comment:
We should add a new test to `KafkaStreamsTest` to check that no directory is created for this case

Reply:
For the above case, `TopologyDiskAccessTest` class is added.  `KafkaStreamsTest` tests only the KafkaStreams logic (It doesn't creates any topology).

Reply:
> KafkaStreamsTest tests only the KafkaStreams logic

Well, because `KafkaStreams` is responsible to create/or-not-crate the directory, this is part of `KafkaStreams` logic, isn't it? Thus, IMHO we should test this behavior in `KafkaStreamsTest`. Nothing prevents you create any topology in the test for this.

Also, `TopologyDisAccessTest` does not use `KafkaStreams` but `TopologyTestDriver` and thus, does not cover this code path.

Reply:
Agree, moved the test cases to `KafkaStreamsTest`.